### ISSUE TITLE
Manual project import: Fails to detect Github "main" branch

### ISSUE THREAD

**Issue by benjaoming** [readthedocs/readthedocs.org#9367]
## Details

Leaving the "Default branch" field empty does not work, but will start to build `master` branches.

* Read the Docs project URL: https://readthedocs.org/projects/example-sphinx-basic/
* Build URL (if applicable): https://readthedocs.org/projects/example-sphinx-basic/builds/17248339/
* Read the Docs username (if applicable): https://readthedocs.org/profiles/benjaoming/

## Expected Result

* Importing a Github project, should immediately start building the default branch -- and if no default branch can be detected, then `main` should be the default
* Help text should mention `main` not `master`

## Actual Result

```
git clone --no-single-branch --depth 50 https://github.com/readthedocs-examples/example-sphinx-basic .
git checkout --force master
error: pathspec 'master' did not match any file(s) known to git
```

![image](https://user-images.githubusercontent.com/374612/175044669-80bc3088-a2b6-4c8b-a757-a6353a5c8c43.png)

**Comment by benjaoming** [comment]
In addition to changing the help-text so it isn't directly misleading, another possible work-around would be to hard-code the initial value of the "Default branch" to `main` for manual imports? So it will always appear like this, forcing the user to take action?

![image](https://user-images.githubusercontent.com/374612/175061282-f37727d5-ad3b-4dd7-9940-9712187cf056.png)

**Comment by stsewd** [comment]
We could also try setting the default branch if the checkout step fails and the default version isn't set, something around these lines

https://github.com/readthedocs/readthedocs.org/blob/86befd21a764518efeefb2e5cd6bdfde3a0f8166/readthedocs/vcs_support/backends/git.py#L326-L329

**Comment by humitos** [comment]
I think we shouldn't guess the default branch or try anything customized on Read the Docs. Why not just _default to the branch that git defaults by default_? I mean, we should remove the step ` git checkout --force master` when we don't have a default branch manually defined. Removing that step, everything should work. Git will use the branch cloned by default when doing `git clone`. That's it.

**Comment by humitos** [comment]
I got another customer hitting this problem. This bug blocks them when importing projects manually (e.g. because they are using Azure DevOps) and generates a bad UX at onboarding.

**Comment by agjohnson** [comment]
> Why not just default to the branch that git defaults by default?

This seems like a good approach. Do we need that text input value to establish the `latest` Version object though?

**Comment by benjaoming** [comment]
I tried the method in [https://stackoverflow.com/a/50056710/405682](this StackOverflow solution) and I like it.

```
git remote show [your_remote] | sed -n '/HEAD branch/s/.*: //p'
```

I would propose to parse the `git remote show` output in Python instead of `sed`.

Tested with:

* github.com :heavy_check_mark: 
* gitlab.com :heavy_check_mark: 
* my own private SSH bare git repo :heavy_check_mark: 
* bitbutcket.org :heavy_check_mark: 

This seems to work with:

* Any currently checked out local branch
* Doesn't matter if remote is `ssh` or `https`
* Running the command needs to connect to the upstream repo - if ssh credentials do not work, the command fails

**NOTICE**

```
git help remote
# ...
       show
           Gives some information about the remote <name>.

           With -n option, the remote heads are not queried first with git ls-remote <name>; cached information is used instead.
```

We can add the option `-n` to cache. This makes the command extremely fast, and since it's a freshly cloned repo, perhaps that's what we want?

The reason I did these tests was a remark on StackOverflow saying _The default branch is a github thing, not a git thing_ but I don't think anymore that the above method has anything specific to do with GitHub.

**Comment by FrostyX** [comment]
The "default branch" field is not a text field anymore, it is a select now. Which is a bit problem, when the `main` branch is missing. I ended up right-clicking to inspect and changed the 

```html
<option value="master" selected="">master</option>
```

to

```html
<option value="main" selected="">main</option>
```

In case anyone needs to workaround the issue as well.

**Comment by benjaoming** [comment]
That workaround seems good @FrostyX - but a fix has also been committed in #9424 and is scheduled to be rolled out at the next release, probably early next week :tada: If you happen to encounter any further issues, please let us know :+1:
